.github/workflows/release.yml

```diff
@@ -1,98 +1,165 @@
-name: Release [WIP]
+name: Release 3 - Publish Release
+run-name: Publish Release ${{ inputs.version }}
 
 on:
-  push:
-    tags:
-    - 'v*'
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Metabase version (e.g. v0.46.3)'
+        type: string
+        required: true
+      commit:
+        description: 'A full-length commit SHA-1 hash'
+        required: true
```

Comment:
I feel pretty strongly about making this trigger on workflow dispatch, this unlocks a lot of flexiblity for us: e.g.

- We can build artifacts on Monday, take time to test them and make sure everything is great about them, and then trigger the release on Wednesday when we're happy
- We can build urgent security fixes artifacts and release them without waiting for the full round of CI checks to pass